sql/sql_parse.cc

```diff
@@ -1993,6 +2019,46 @@ bool dispatch_command(THD *thd, const COM_DATA *com_data,
   DBUG_ASSERT(thd->open_tables == NULL ||
               (thd->locked_tables_mode == LTM_LOCK_TABLES));
 
+  /* Update user statistics */
+  if (unlikely(opt_userstat)) {
```

Comment:
There is a race condition in that opt_userstat can change its value between the two sections. So in the second section I'd only check for start_* timestamp initialization